main/src/libraries/wrdac/src/clients/icubClient.cpp

```diff
@@ -1047,5 +1060,10 @@ SubSystem_KARMA* ICubClient::getKARMA()
     if (subSystems.find(SUBSYSTEM_KARMA) == subSystems.end())
         return NULL;
     else
-        return (SubSystem_KARMA*)subSystems[SUBSYSTEM_KARMA];
+    {
+        SubSystem_KARMA *karma = (SubSystem_KARMA*)subSystems[SUBSYSTEM_KARMA];
+        karma->setRobot(robot);

```

Comment:
I wouldn't call `setRobot` every time `getKarma` is called. Instead, I propose to modify the constructor so it accepts the robot name as attribute, and pass the robot name to the constructor here: `https://github.com/towardthesea/wysiwyd/blob/02a8038bf5d864d8410d6c7fafdccc4617db1c4c/main/src/libraries/wrdac/src/clients/icubClient.cpp#L135`.
Otherwise, this is significant overhead I think.